packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ClientModelPlugin.cs

```diff
@@ -25,6 +26,21 @@ public class ClientModelPlugin : CodeModelPlugin
 
         public override ExtensibleSnippets ExtensibleSnippets { get; }
 
+        /// <summary>
+        /// Returns the serialization type providers for the given model type provider.
+        /// </summary>
+        /// <param name="provider">The model type provider.</param>
+        public override TypeProvider[] GetSerializationTypeProviders(ModelTypeProvider provider)
+        {
+            // Add JSON serialization type provider
+            if (provider.Usage.HasFlag(InputModelTypeUsage.Json))
+            {
+                return new TypeProvider[] { new JsonMrwSerializationTypeProvider(provider) };
+            }
+
+            return Array.Empty<TypeProvider>();
+        }
```

Comment:
how could this be multiple typeproviders?
Json serialization will write 4 methods, Json+Xml serialization also writes 4 methods, just the implementation of it changes. Xml serialization only we need to write 2 methods (from IPersistableModel<T>)
It is not like something that we have one partial class for one serialization format.
If we are writing something like:
```csharp
public partial class Foo : IJsonModel<Foo>
{
	// this class implements the interfaces and calls other functions for formats
	BinaryData IPersistableModel<Foo>.Write(MRWOptions options)
	{
		switch (options.Format)
		{
			case "J":
				return WriteJson(options);
			case "X":
				return WriteXml(options);
			default:
				throw;
		}
	}
}

// Foo.Json.Serialization.cs
public partial class Foo
{
	private void WriteJson();
}
// Foo.Xml.Serialization.cs
public partial class Foo
{
	private void WriteXml();
}
```

In the above way I think multiple typeproviders make sense, but from the perspective of writing the methods we need, we do not need multiple therefore here we should not return an array.

Reply:
I don't think MRW will have 2 implementations that is a good point it supports both xml and json.  The thought here was what if you wanted XElement and Newtonsoft, that is two implementations that shouldn't necessarily be lumped together so an array of providers was the idea.

@JoshLove-msft is still finalizing the extension point so this can change in the future (and likely will) but wanted to shape it this way in this iteration to keep that concept in mind that someone might want multiple serialization implementations